### ISSUE TITLE
**stop_if_no_decrease_hook** did not work during training?

### ISSUE THREAD

**Issue by alansplaza** [tensorflow/tensorflow#40443]
```
    early_stopping_hook = tf.estimator.experimental.stop_if_no_decrease_hook(
                                deepfm_model,
                                metric_name='loss',
                                max_steps_without_decrease=13,
                                min_steps=5)
    training_hooks.append(early_stopping_hook)

    tf.estimator.train_and_evaluate(
        estimator=deepfm_model,
        train_spec=tf.estimator.TrainSpec(train_input_fn, hooks=training_hooks),
        eval_spec=tf.estimator.EvalSpec(eval_input_fn, steps=100)#, exporters=best_exporter)
    )
```

Using the code, as the params define, the evaluation should be processed every 13 training steps and the training process should be ended if the loss did not decrease. 

But it keep training and did not do evaluation every 13 training steps.

**Comment by amahendrakar** [comment]
@alansplaza,
In order to expedite the trouble-shooting process, could you please provide the complete code to reproduce the issue reported here and also the TensorFlow version you are using. Thanks!
